When new namespace is created in memory [Namespace#register()](https://github.com/lucmoreau/ProvToolbox/blob/ca05849c88a84aae736355569a8bdae78c68b072/modules-core/prov-model/src/main/java/org/openprovenance/prov/model/Namespace.java#L124) is used. This method does no validation of the URI.
Therefore, Java allows document with URI with space to be created in memory. 

When serializing PROVN format, there is no validation of URI.
But this is not wrong, since the serializer should not expect that PROV document with invalid URI can be generated by Java. Its the model where invalid URI can be generated, where the issue starts.


PROVN deserialization in python is not implemented yet. 
As stated in [ProvPy features](https://pypi.org/project/prov/), "In-memory classes for PROV assertions, which can then be output as PROV-N", implying that only PROV-N serialization is supported.
It has been worked on but abandoned as you can see in this issue on library's [github](https://github.com/trungdong/prov/issues/122).


